package.json

```diff
@@ -1,37 +1,41 @@
     "test": "yarn run clean && yarn run setup-paths && yarn run test:node-deps && yarn run mocha",
+    "test:with-optimized-types": "yarn run clean && yarn run setup-paths && yarn run test:node-deps-optimized && yarn run mocha",
     "test:watch": "cross-env BABEL_ENV='browser' node -- scripts/test-build-watch.js --with-dependency-tracking .tmp/test/tests.js",
     "test:web-deps": "concurrently 'yarn run test:copy-web-deps' 'yarn run schema:transform-to-types'",
-    "test:node-deps": "yarn run schema:transform-to-types && cross-env BABEL_ENV='node' node -- scripts/test-build-node.js --with-dependency-tracking .tmp/test/node-tests.js",
+    "test:node-deps": "yarn run schema:transform && cross-env BABEL_ENV='node' node -- scripts/test-build-node.js --with-dependency-tracking .tmp/test/node-tests.js",
+    "test:node-deps-optimized": "yarn run schema:transform && cross-env BABEL_ENV='node' node -- scripts/test-build-node.js --with-optimized-types .tmp/test/node-tests.js",
     "test:copy-web-deps": "cp test/index.html node_modules/mocha/mocha.* node_modules/jquery/dist/jquery.slim.min.js .tmp/test/ && cp node_modules/expect.js/index.js .tmp/test/expect.js",
     "test:livereload-server": "node scripts/livereload.js",
     "mocha": "mocha -u tdd $(yarn run mocha:reporter-args 2>&1 >/dev/null) .tmp/test/node-tests.js",
-    "mocha:reporter-args": "test -n \"${CI}\" && >&2 echo --reporter xunit --reporter-options output=$CIRCLE_TEST_REPORTS/junit/mocha.xml || true",
+    "mocha:reporter-args": "test -n \"${CI}\" && >&2 echo --reporter xunit --reporter-options output=$CIRCLE_TEST_REPORTS/junit/mocha-$RANDOM.xml || true",
     "lint": "eslint --max-warnings 0 -c .eslintrc.json $(yarn run lint:reporter-args 2>&1 >/dev/null) src/ test/",
     "lint:reporter-args": "test -n \"${CI}\" && >&2 echo -o $CIRCLE_TEST_REPORTS/junit/eslint.xml -f junit",
     "print-start-message": "wait-on file:.tmp/test/index.html file:.tmp/test/tests.js tcp:35729 tcp:4200 file:types.js && echo \"\n\n⚡️⚡️⚡️  Good to go at http://localhost:4200 ⚡️⚡️⚡️\"",
     "schema:refresh": "rimraf types.js && yarn run schema:fetch && yarn run schema:transform-to-types",
     "schema:fetch": "graphql-js-schema-fetch --url 'https://graphql.myshopify.com/api/graphql' --header 'Authorization: Basic MzUxYzEyMjAxN2QwZjJhOTU3ZDMyYWU3MjhhZDc0OWM=' | jq '.' > schema.json",
-    "schema:transform-to-types": "graphql-js-schema --schema-file schema.json --outdir . --schema-bundle-name 'Types' --bundle-only"
+    "schema:transform": "concurrently 'yarn run schema:transform-to-types' 'yarn run schema:transform-to-optimized-types'",
+    "schema:transform-to-types": "graphql-js-schema --schema-file schema.json --outdir . --schema-bundle-name 'Types' --bundle-only",
+    "schema:transform-to-optimized-types": "graphql-js-schema $(node scripts/profiled-types-to-whitelist-args.js) --schema-file schema.json --outdir . --schema-bundle-name 'OptimizedTypes' --bundle-only"
```

Comment:
Minimist supports it transparently. This worked without any changes:

```bash
"schema:test": "graphql-js-schema --whitelist-type={$(cat types.csv)} --schema-file schema.json --outdir . --schema-bundle-name 'OptimizedTypes' --bundle-only"
```

I'm good with the argument about wanting JSON for other reasons though if it will be used for other purposes 👍